build-system/tasks/check-owners.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview This file implements the `gulp check-owners` task, which checks
+ * all OWNERS files in the repo for correctness, as determined by the parsing
+ * API provided by the AMP owners bot.
+ */
+
+'use strict';
+
+const deglob = require('globs-to-files');
+const fs = require('fs-extra');
+const JSON5 = require('json5');
+const log = require('fancy-log');
+const path = require('path');
+const {cyan, red, green} = require('ansi-colors');
+const {isTravisBuild} = require('../common/travis');
+
+const rootDir = path.dirname(path.dirname(__dirname));
+
+/**
+ * Checks OWNERS files for correctness using the amp-github-apps API.
+ * The cumulative result is returned to the `gulp` process via process.exitCode
+ * so that all OWNERS files can be checked / fixed.
+ */
+async function checkOwners() {
+  const filesToCheck = deglob.sync(['**/OWNERS']);
+  filesToCheck.forEach(file => {
+    const relativePath = path.relative(rootDir, file);
+    checkFile(relativePath);
+  });
```

Comment:
Since this function is `async`, it might make sense to tag `checkFile` as `async` too since it'll be making an API call. Once we make it `async`, then we'd need an `await` here; the anonymous function will become `async file => { ... }`, and we'll hit the issue of `forEach` not playing very nicely with async functions (it will just execute them all and not wait for the promises to resolve). To avoid (me) missing that and possibly introducing a bug in the future, might be worth adding all the `async`/`await`s and replacing the loop either with an explicit `for` or with `Promises.all(filesToCheck.map(async file => { ... await checkFile(...); });`

Reply:
Done.